ppx/reason_react_ppx.ml

```diff
@@ -321,18 +321,15 @@ let rec recursivelyMakeNamedArgsForExternal list args =
            | _label, Some type_, Some _ -> type_
            (* ~foo: option(int)=? *)
            | ( _label,
-               Some
```

Comment:
This case should remain here and add a discrimination when it's not `*predef*`, otherwise the wrapping from the user disapears

Reply:
In regular OCaml, and I'll include a [playground link](https://melange.re/v2.0.0/playground/?language=Reason&code=bGV0IG15Rm4gPSAofmE6IG9wdGlvbihpbnQpPT8sICgpKSA9PiBPcHRpb24udmFsdWUoYSwgfmRlZmF1bHQ9MCk7CgpleHRlcm5hbCBteUZuRXh0ZXJuYWw6ICh%2BYTogaW50PT8sIHVuaXQpID0%2BIGludCA9ICJkb2VzbkV4aXN0IjsKCmxldCAoKSA9IHsKICBsZXQgXzogaW50ID0gbXlGbih%2BYT0zLCAoKSk7CiAgbGV0IF86IGludCA9IG15Rm5FeHRlcm5hbCh%2BYT0zLCAoKSk7CgogIGxldCBfOiBpbnQgPSBteUZuKCk7CiAgbGV0IF86IGludCA9IG15Rm5FeHRlcm5hbCgpOwoKICBsZXQgXzogaW50ID0gbXlGbih%2BYT0%2FU29tZSgzKSwgKCkpOwogIGxldCBfOiBpbnQgPSBteUZuRXh0ZXJuYWwofmE9P1NvbWUoMyksICgpKTsKICAoKTsKfTsK&live=off), 

there's the following distinction:

```reason
let myFn = (~a: option(int)=?, ()) => Option.value(a, ~default=0);

external myFnExternal: (~a: int=?, unit) => int = "doesnExist";
```

these have the same signature. my change makes our PPX consistent with the default OCaml behavior. I believe this is fixing a bug. What use cases am I missing?